controllers/virtualmachine/storagepolicyusage/webhookconfiguration/webhookconfiguration_controller.go

```diff
@@ -0,0 +1,133 @@
+
+	r := NewReconciler(
+		ctx,
+		mgr.GetClient(),
+		ctrl.Log.WithName("controllers").WithName(controlledTypeName),
+		record.New(mgr.GetEventRecorderFor(controllerNameLong)),
+	)
+
+	c, err := controller.New(controllerNameShort, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	cache, err := pkgmgr.NewNamespacedCacheForObject(mgr, &ctx.SyncPeriod, controlledType)
+	if err != nil {
+		return err
+	}
+
+	return c.Watch(source.Kind(cache, controlledType, &handler.TypedEnqueueRequestForObject[*admissionv1.ValidatingWebhookConfiguration]{},
+		predicate.TypedFuncs[*admissionv1.ValidatingWebhookConfiguration]{
+			CreateFunc: func(e event.TypedCreateEvent[*admissionv1.ValidatingWebhookConfiguration]) bool {
+				return e.Object.GetName() == spqutil.ValidatingWebhookConfigName
+			},
+			UpdateFunc: func(e event.TypedUpdateEvent[*admissionv1.ValidatingWebhookConfiguration]) bool {
+				return e.ObjectOld.GetName() == spqutil.ValidatingWebhookConfigName
+			},
+			DeleteFunc: func(e event.TypedDeleteEvent[*admissionv1.ValidatingWebhookConfiguration]) bool {
+				return false
+			},
+			GenericFunc: func(e event.TypedGenericEvent[*admissionv1.ValidatingWebhookConfiguration]) bool {
+				return false
+			},
+		},
+		kubeutil.TypedResourceVersionChangedPredicate[*admissionv1.ValidatingWebhookConfiguration]{},
+	))
+}
+
+func NewReconciler(ctx context.Context, client client.Client, logger logr.Logger, recorder record.Recorder) *Reconciler {
+
+	return &Reconciler{
+		Context:  ctx,
+		Client:   client,
+		Logger:   logger,
+		Recorder: recorder,
+	}
+}
+
+// Reconciler reconciles a ValidatingWebhookConfiguration object.
+type Reconciler struct {
+	client.Client
+	Context  context.Context
+	Logger   logr.Logger
+	Recorder record.Recorder
+}
+
+// +kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=validatingwebhookconfiguration,verbs=get;list;watch
+
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
+	ctx = pkgcfg.JoinContext(ctx, r.Context)
+
+	if req.Name != spqutil.ValidatingWebhookConfigName {
+		r.Logger.Error(nil, "Reconcile called for unexpected object", "name", req.Name)
+		return ctrl.Result{}, nil
+	}
+	return ctrl.Result{}, r.ReconcileNormal(ctx, req)
+}
+
+func (r *Reconciler) ReconcileNormal(ctx context.Context, req ctrl.Request) error {
+	r.Logger.Info("Reconciling validating webhook configuration", "name", req.Name)
+
+	caBundle, err := spqutil.GetWebhookCABundle(ctx, r.Client)
+	if err != nil {
+		return err
+	}
+
+	spuList := &spqv1.StoragePolicyUsageList{}
+	if err := r.Client.List(ctx, spuList); err != nil {
+		return fmt.Errorf("unable to list StoragePolicyUsage objects: %w", err)
+	}
+
+	for _, spu := range spuList.Items {
+		if spu.Spec.ResourceExtensionName == spqutil.StoragePolicyQuotaExtensionName {
+			if !bytes.Equal(spu.Spec.CABundle, caBundle) {
+				spu.Spec.CABundle = caBundle
+				if err := r.Client.Update(ctx, &spu); err != nil {
```

Comment:
`Update` should only be used in rare circumstances. Please use `Patch` instead. There is an example [here](https://github.com/vmware-tanzu/vm-operator/blob/39cdfbfab5f9174f1c56a04c0c9387f46d7823d1/controllers/virtualmachine/storagepolicyusage/storagepolicyusage_controller.go#L158-L172), although you will want to use the regular client, not the status client.